roles/rsyslog/tasks/main_core.yml

```diff
@@ -21,7 +21,8 @@
 
     - name: Reset original confs - logging package is absent
       package:
-        name: "{{ __rsyslog_base_packages }}"
+        # only remove rsyslog itself, not other packages like iproute
+        name: "{{ __rsyslog_base_packages | select('search', 'rsyslog') }}"
         state: absent
         use: "{{ (__logging_is_ostree | d(false)) |
```

Comment:
This is bad - not sure what I was thinking here - if on an ostree/image mode system, you cannot remove packages - which means this block can likely be skipped `when: __logging_is_ostree | d(false)` - which means removing the configuration and restoring it will not work on ostree either

Reply:
For non-ostree systems, a better approach would be to not remove the package, but remove/backup the config file and reinstall the package. That should avoid the reverse requires etc. trouble.

For ostree, the only thing that comes to my mind is to download the rpm and extract the config, but that's "ugh", and on the final system we may not even have `dnf`. OTOH, on a bootc system it's not likely that someone *both* uses lsr/logging *and* put in a customized rsyslog.conf first, so I think we could afford to just skip over this on ostree (I'm not sure if that's what it's doing, naïvely the condition reads wrong: it should be *skipped* on ostree, not *only* be run on ostree. But that's apparently what's happening, so this is just weird syntax.

Reply:
@richm for ostree systems we have a way out -- we can grab the original config from /usr/etc/rsyslog.conf :tada: Perhaps we can land this PR first, which improves the situation for rpm-based systems, and I'll look into the ostree path separately when we start testing that?